guides/ci/travis.md

```diff
@@ -0,0 +1,364 @@
+
+Now let’s see an example of using Travis CI with Crystal.
+
+## Build and run specs
+
+### Using `latest` and `nightly`
+
+When [creating a Crystal project](https://crystal-lang.org/reference/using_the_compiler/#creating-a-crystal-project) using `crystal init`, Crystal creates a `.travis.yml` file for us:
+
+```yaml
+# .travis.yml
+language: crystal
+
+crystal:
+  - latest
+  - nightly
+
+script:
+  - crystal spec
+  - crystal tool format --check
+```
+
+Although a basic configuration file, it will let’s use Travis CI. Now, let’s go to Travis CI dashboard to [add the GitHub repository](https://docs.travis-ci.com/user/tutorial/#to-get-started-with-travis-ci) From this moment, Travis CI will run the `specs` against the Crystal compiler (using both `latest` and `nightly` releases)
+
+### Using a specific Crystal release
+
+So, let’s suppose we want to use version [0.31.1](https://github.com/crystal-lang/crystal/releases/tag/0.31.1)
+ of the Crystal compiler. We will be tempting to add `- 0.31.1`. But, as we may read at the [Crystal configuration options](https://docs.travis-ci.com/user/languages/crystal/#configuration-options), we can only use the values `latest` and `nightly`.
+
+So, how do we test against a specific version?
+
+For this we are going to use [Docker](https://www.docker.com/). First we need to add Docker as a service in `.travis.yml`, and then use `docker` from the CLI to run the specs with a specific Crystal version, like this:
+
+```yml
+# .travis.yml
+language: bash
+
+services:
+  - docker
+
+script:
+  - docker run -v $PWD:/src -w /src crystallang/crystal:0.31.1 crystal spec spec/game_of_life_spec.cr
+```
+
+**Note:** We may find the list with the different official [Crystal docker images](https://hub.docker.com/r/crystallang/crystal/tags) at [DockerHub](https://hub.docker.com/r/crystallang/crystal)
+
+**Note:** This is a great trick that will let us use different Crystal versions without the need of actually installing the compiler and without the need of creating a `Dockerfile` configuration file.
+
+### Using `latest`, `nightly` and a specific Crystal release all together!
+
+For using 3 versions of the compiler we are going to use a different configuration file based on [Build Matrix](https://docs.travis-ci.com/user/customizing-the-build#build-matrix)
+Here is the example:
+
+```yaml
+# .travis.yml
+matrix:
+ include:
+   - language: crystal
+     crystal:
+       - latest
+     script:
+       - crystal spec
+
+   - language: crystal
+     crystal:
+       - nightly
+     script:
+       - crystal spec
+
+   - language: bash
+     services:
+       - docker
+     script:
+       - docker run -v $PWD:/src -w /src crystallang/crystal:0.31.1 crystal spec spec/game_of_life_spec.cr
+```
+
+## Using external dependencies
+
+Before presenting some examples, it is important to mention that there are many ways to achieve this and **it will heavily depend on our development workflow**.
```

Comment:
I think some intro is missing in this part. The goal is to install native libraries that might be linked by the app. `sqlite3` could be a better example than `git`. ( `sqlite3 --version` works ;-) )

And maybe a clarification, or short previous section, to mention that shards dependencies are installed directly in the `install` step thanks to the `language: crystal` support. Otherwise talking about dependencies could be misleading.